src/query/graphite/native/builtin_functions.go

```diff
@@ -1000,6 +1000,43 @@ func integral(ctx *common.Context, input singlePathSpec) (ts.SeriesList, error)
 	return r, nil
 }
 
+// integralByInterval will do the same as integral funcion, except it resets the total to 0
+// at the given time in the parameter “from”. Useful for finding totals per hour/day/week.
+func integralByInterval(ctx *common.Context, input singlePathSpec, intervalString string) (ts.SeriesList, error) {
+	intervalUnit, err := common.ParseInterval(intervalString)
+	if err != nil {
+		return ts.NewSeriesList(), err
+	}
+	results := make([]*ts.Series, 0, len(input.Values))
+	for _, series := range input.Values {
+		stepsPerInterval := intervalUnit.Milliseconds() /  int64(series.MillisPerStep())
+		var stepCounter int64 = 0
+
+		outvals := ts.NewValues(ctx, series.MillisPerStep(), series.Len())
+		var currentSum float64
```

Comment:
It is more Go-ish to combine these var assignments altogether into a multi line var statement (and leave the zero values unassigned, i.e. `stepCounter` and `currentSum`):
```go
var (
  stepsPerInterval = intervalUnit.Milliseconds() /  int64(series.MillisPerStep())
  vals = ts.NewValues(ctx, series.MillisPerStep(), series.Len())
  stepCounter int64
  currentSum float64
)
```

Also "outvals" should either be camel cased (since it's two words) or just one word (i.e. vals).

Reply:
done! thanks for this.